sway/commands/seat/shortcuts_inhibitor.c

```diff
@@ -0,0 +1,86 @@
+#include "log.h"
+#include "sway/commands.h"
+#include "sway/input/seat.h"
+#include "sway/input/input-manager.h"
+#include "util.h"
+
+static struct cmd_results *handle_action(struct sway_seat *seat,
+		const char *action) {
+	struct sway_keyboard_shortcuts_inhibitor *sway_inhibitor = NULL;
+	if (strcmp(action, "disable") == 0) {
+		wl_list_for_each(sway_inhibitor,
+				&seat->keyboard_shortcuts_inhibitors, link) {
+			wlr_keyboard_shortcuts_inhibitor_v1_deactivate(
+					sway_inhibitor->inhibitor);
+		}
+
+		sway_log(SWAY_DEBUG, "Deactivated all keyboard shortcuts inhibitors");
+	} else {
+		sway_inhibitor = keyboard_shortcuts_inhibitor_get_for_focused_surface(seat);
+		if (!sway_inhibitor) {
+			return cmd_results_new(CMD_FAILURE,
+					"No inhibitor found for focused surface");
+		}
+
+		struct wlr_keyboard_shortcuts_inhibitor_v1 *inhibitor =
+			sway_inhibitor->inhibitor;
+		bool inhibit = parse_boolean(action, inhibitor->active);
+		if (inhibit) {
+			wlr_keyboard_shortcuts_inhibitor_v1_activate(inhibitor);
+		} else {
+			wlr_keyboard_shortcuts_inhibitor_v1_deactivate(inhibitor);
+		}
+
+		sway_log(SWAY_DEBUG, "%sctivated keyboard shortcuts inhibitor",
+				inhibit ? "A" : "Dea");
+	}
+
+	return cmd_results_new(CMD_SUCCESS, NULL);
+}
+
+// shortcuts_inhibitor [enable|disable|active|inactive|toggle]
+struct cmd_results *seat_cmd_shortcuts_inhibitor(int argc, char **argv) {
+	struct cmd_results *error =
+		checkarg(argc, "shortcuts_inhibitor", EXPECTED_EQUAL_TO, 1);
+	if (error) {
+		return error;
+	}
+
+	struct seat_config *sc = config->handler_context.seat_config;
+	if (!sc) {
+		return cmd_results_new(CMD_FAILURE, "No seat defined");
+	}
+
+	if (strcmp(argv[0], "enable") == 0) {
```

Comment:
When used at runtime, this should probably also deactivate any active inhibitor

Reply:
I had thought about *activating* all existing inhibitors when enabling the overall functionality but that seems overzealous to me. I agree with @RedSoxFan that disabling the functionality should also deactivate all active inhibitors for good measure and consistency but why should enabling it do the same?

Reply:
Oh, sorry, I misread your code. I agree, only `disable` should be a special-case.

Reply:
Ah, well, if it's easily misread then maybe control flow is too complicated now. Do you have a suggestion where I maybe didn't fully understand @RedSoxFan's suggestion of doing the disabling int `seat_apply_config()`?